### ISSUE TITLE
Bulk editor for orders fails to move order to "done"

### ISSUE THREAD

**Issue by fooku18** [shopware/platform#3524]
### PHP Version

7.4

### Shopware Version

6.4.20.2

### Expected behaviour

When using the bulk editor for orders, the transition fails when the orders are moved to "done". Within Shopware, I have extended custom state_machine_states for the `order.state` state machine. The flow is as follows:

**in progress -> in manufacturing -> manufacturing done -> completed**

I have added a flow rule that moves the order to "completed", when the order state enters "manufacturing done"
![Screenshot from 2024-01-18 03-49-39](https://github.com/shopware/shopware/assets/22157117/dbb2aed0-e4cc-4dfc-b05e-cebf55ab5f5d)

When using the bulk editor and moving all orders to "manufacturing done", I expect the flow rule to trigger and move the order to "completed"
![Screenshot from 2024-01-18 03-50-55](https://github.com/shopware/shopware/assets/22157117/c53f651c-3055-4620-a073-2d2d036915b4)

### Actual behaviour

When triggering the bulk editor for the orders, it works for some of the orders but fails with a deadlock while trying to update the stock for a product:

```
2024/01/18 02:51:01 [error] 2242#2242: *612 FastCGI sent in stderr: "PHP message: SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transactionPHP message: #0 /var/www/html/app/shopware/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php(119): Doctrine\DBAL\Driver\PDO\Exception::new(Object(PDOException))
#1 /var/www/html/app/shopware/vendor/shopware/core/Framework/DataAbstractionLayer/Doctrine/RetryableQuery.php(74): Doctrine\DBAL\Driver\PDOStatement->execute(Array)
#2 /var/www/html/app/shopware/vendor/shopware/core/Framework/DataAbstractionLayer/Doctrine/RetryableQuery.php(122): Shopware\Core\Framework\DataAbstractionLayer\Doctrine\RetryableQuery->Shopware\Core\Framework\DataAbstractionLayer\Doctrine\{closure}()
#3 /var/www/html/app/shopware/vendor/shopware/core/Framework/DataAbstractionLayer/Doctrine/RetryableQuery.php(78): Shopware\Core\Framework\DataAbstractionLayer\Doctrine\RetryableQuery::retry(Object(Doctrine\DBAL\Connection), Object(Closure), 1)
#4 /var/www/html/app/shopware/vendor/shopware/core/Content/Product/DataAbstractionLayer/StockUpdater.php(402)
```

analyzing the deadlock further indicates that this is due to a lock within the update of the products stock:

```
------------------------
LATEST DETECTED DEADLOCK
------------------------
2024-01-18 02:51:01 0x7fcf3cda3700
*** (1) TRANSACTION:
TRANSACTION 4717251, ACTIVE 0 sec starting index read
mysql tables in use 43, locked 43
LOCK WAIT 11 lock struct(s), heap size 1128, 6 row lock(s), undo log entries 2
MySQL thread id 310, OS thread handle 140528055584512, query id 108705 172.18.0.2 db Updating
UPDATE product SET stock = stock + '-1' WHERE id = '????tKŋ????.' AND version_id = '???jK¾K??u,4%'
*** (1) WAITING FOR THIS LOCK TO BE GRANTED

[...]

*** (2) TRANSACTION:
TRANSACTION 4717249, ACTIVE 0 sec starting index read
mysql tables in use 43, locked 43
12 lock struct(s), heap size 1128, 7 row lock(s), undo log entries 3
MySQL thread id 309, OS thread handle 140528055891712, query id 108706 172.18.0.2 db Updating
UPDATE product SET available_stock = stock - '193', sales = '538', updated_at = '2024-01-18 02:51:01.928' WHERE id = '????tKŋ????.'
*** (2) HOLDS THE LOCK(S)
```

The behaviour kicks in when moving more than 2 order of the same product via the bulk editor. When only applying 1 or 2, it generally works (could not verify one case where it was crashing). Nevertheless, due to the intention of the bulk editor, it would be great if this would work for more than 2 orders in parallel.

### How to reproduce

This should be also reproducible by adding a flow rule that moves the order from whatever order state to "completed" and by moving multiple orders via the bulk editor.

**Comment by MarcelSchmaeing** [comment]
Hi,

thx for your issue we will put it into our new evaluation process and have a look at it again.

**Comment by shopwareBot** [comment]
[public] Automated response: This issue is linked to the internal issue [https://shopware.atlassian.net/browse/NEXT-37901](https://shopware.atlassian.net/browse/NEXT-37901).

 

[created from NEXT-37901, comment 488763]

**Comment by shopwareBot** [comment]
[public] Hello,

 Within our company roadmap and work capacity, we try to address each bug or improvement request but admit that not each one will be resolved. To continue our culture of honesty and openness, we are closing this issue to focus on our roadmap on behalf of all Shopware users.

 If you feel like this issue is still impacting you, please create a new issue and let us know. Thank you again for your valuable feedback.



[created from NEXT-37901, comment 549969]
